### PR TITLE
fix oauth2 integration allowing for better handling of JWT token auth…

### PR DESCRIPTION
**Pull request by yzhivkov**
…orization and oauth2 access token (#383)

- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/jhipster-registry/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-registry/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: src/main/java/io/github/jhipster/registry/security/oauth2/AuthorizationHeaderUtil.java [hunk 1]
```diff
@@ -1,43 +1,163 @@
...
+    private OAuth2AccessTokenResponse toOAuth2AccessTokenResponse(OAuthIdpTokenResponseDTO oAuthIdpResponse) {
+        Map<String, Object> additionalParameters = new HashMap<>();
+        additionalParameters.put("id_token", oAuthIdpResponse.getIdToken());
+        additionalParameters.put("not-before-policy", oAuthIdpResponse.getNotBefore());
+        additionalParameters.put("refresh_expires_in", oAuthIdpResponse.getRefreshExpiresIn());
+        additionalParameters.put("session_state", oAuthIdpResponse.getSessionState());
+        return OAuth2AccessTokenResponse.withToken(oAuthIdpResponse.getAccessToken())
+            .expiresIn(oAuthIdpResponse.getExpiresIn())
+            .refreshToken(oAuthIdpResponse.getRefreshToken())
+            .scopes(Pattern.compile("\\s").splitAsStream(oAuthIdpResponse.getScope()).collect(Collectors.toSet()))
```
Comment (gmarziou) [inline src/main/java/io/github/jhipster/registry/security/oauth2/AuthorizationHeaderUtil.java]: Pattern compilation should probably be extracted to a static filed

**Comment by gmarziou** [comment]
Thanks @yzhivkov for this contribution, I should have started by this :)

**Comment by aanno2** [comment]
Because of my own pull request #390, I closely looked into the OAuth2 implementation of v5.0.2. 

**I strongly vote for prompt merging of this request** as it fixes some important issues, and it implements refreshing of OAuth2 tokens.

(BTW, it does not conflict with #390...)

**Comment by mraible** [comment]
I'm not comfortable merging this until it's tested against Okta. The last time I ran JHipster Registry with JHipster 6 (via Docker Compose), it worked fine for me. I'll admit, I didn't do rigorous testing. However, I was able to log in and Eureka registration worked with both Keycloak and Okta.

I'll make some time to look into it this weekend.

**Comment by pascalgrimaud** [comment]
everything should work for OAuth2 in the last version of Registry

**Pull request closed without merging**